src/odemis/gui/cont/tabs.py

```diff
@@ -1716,6 +1716,16 @@ def __init__(self, name, button, panel, main_frame, main_data):
         self._settings_controller.setter_temporalspec_bck_file = self.set_temporalspec_background
         self._settings_controller.setter_spec_file = self.set_spec_comp
 
+        if main_data.role is None:
+            # HACK: Move the logo to inside the FileInfo bar, as the tab buttons
+            # are hidden when only one tab (ie, no backend).
+            capbar = panel.fp_fileinfo._caption_bar
+            logo = main_frame.logo
+            assert main_frame.pnl_tabbuttons.GetSizer().Detach(logo)
+            logo.Reparent(capbar)
+            # Move to the right of the bar (before the icon)
+            logo.SetPosition((capbar.Size[0] - logo.Size[0] - 20 - 16 - 6, 8))
```

Comment:
You could add a small comment where the  - 20 - 16 - 6 numbers are coming from

Reply:
Indeed, that looks a little too magical... -> added extra comment.